### ISSUE TITLE
Modal: trap focus inside opened modal

### ISSUE THREAD

**Issue by pkozlowski-opensource** [ng-bootstrap/ng-bootstrap#642]
We need to make sure that focus can't "escape" the opened modal (clicks outside, tab, shift+tab etc.).

**Comment by yogsadafal** [comment]
@pkozlowski-opensource any updates on this issue? when this will be available?

**Comment by dholovin** [comment]
also curious about any updates.
Thanks

**Comment by AntonLebiodkin** [comment]
+1
any updates?

**Comment by warapitiya** [comment]
updates?

**Comment by trilokkrothapalli** [comment]
any updates ?

**Comment by jpeterson88** [comment]
Any updates?

**Comment by dantheman999301** [comment]
Not sure if I'm doing something wrong here but I've upgraded to 3.2.0 and can still tab out of modals.

**Comment by MarijkeM** [comment]
Using version 5.1.0 and still able to tab outside of a modal
@pkozlowski-opensource, could you reopen this issue?

**Comment by dantheman999301** [comment]
@MarijkeM as a work around I vaguely remember pulling in some of the Angular Material CDK and using the focus trap directly, which did work.

https://material.angular.io/cdk/a11y/overview

**Comment by nicofilizzola** [comment]
The solution that I implemented is the following:

1. Create a trapFocus directive as explained in [this article](https://medium.com/allenhwkim/angular-trap-focus-with-in-a-block-79b7572f23c2) 
2. In the modal's template, wrap the modal's content by a div using that same directive as follows (this shouldn't affect the layout but keep an eye on it anyway):
`<ng-template #modal let-modal>
      <div trapFocus>
              ...
      </div>
</ng-template>`

The modal component should be able to do this by default though, so I encourage the developers to integrate it.
